app/integration_test/app_test.dart

```diff
@@ -134,3 +134,46 @@ class _UserCredentials {
   /// The password of the user.
   final String password;
 }
+
+/// Waits for a widget identified by [finder] to be present in the widget tree.
+///
+/// This function can be useful when there is no load indicator (if there were,
+/// `await tester.pumpAndSettle()` would wait until the load animation was
+/// finished) and the widget tree is not yet ready to be tested.
+///
+/// The function repeatedly calls `tester.pumpAndSettle()` and then delays for a
+/// short duration, checking at each iteration if the widget identified by
+/// [finder] is present in the widget tree. This continues until the widget is
+/// found or the [timeout] duration has passed, whichever occurs first.
+///
+/// Throws an [Exception] if the [timeout] duration passes without the widget
+/// being found in the widget tree.
+///
+/// Workaround for https://github.com/flutter/flutter/issues/88765.
+///
+/// ## Example
+///
+/// ```dart
+/// await waitFor(tester, find.text('Hello, World'));
+/// ```
+///
+/// ## Exceptions
+///
+/// Throws an [Exception] if the [timeout] is reached before the widget is
+/// found.
+Future<void> waitFor(
+  WidgetTester tester,
+  Finder finder, {
+  Duration timeout = const Duration(seconds: 70),
+}) async {
+  final end = tester.binding.clock.now().add(timeout);
+
+  do {
+    if (tester.binding.clock.now().isAfter(end)) {
+      throw Exception('Timed out waiting for $finder');
+    }
+
+    await tester.pumpAndSettle();
```

Comment:
Wouldn't this hang indefinitely if there is an endless/repeating Animation playing?

Reply:
Since it's `do while` wouldn't it mean that with an endless animation we would never even come to the `finder.evaluate()` part since it would hang at `finder.pumpAndSettle`?

Reply:
Yes, in the case we have an endless animation, we can't use `tester.pumpAndSettle()` for this place in the app anymore

Reply:
> Since it's do while wouldn't it mean that with an endless animation we would never even come to the finder.evaluate()

Even without a `while` loop and just `tester.pumpAndSettle()` would an endless animation never complete the `tester.pumpAndSettle` (only when the default timeout of 10 minutes for `pumpAndSettle`() hits).

Reply:
I mean since it works for this we can merge it like this if u want, but couldn't we just replace `pumpAndSettle`  with `pump` since its looped anyways?

Reply:
Yes, this could work. Locally it worked. Let's try this.